src/CMakeLists.txt

```diff
@@ -2,7 +2,74 @@
 include(SwiftSupport)
 include(DTrace)
 
-add_library(dispatch
+if(ENABLE_SWIFT)
+    set(dispatch_targets dispatch dispatch_static dispatch_objects)
+else()
+   set(dispatch_targets dispatch)
+endif()
+
+function(dispatch_add_library)
+    # Swift builds both shared and static libraries
+    if(ENABLE_SWIFT)
+        add_library(dispatch_objects OBJECT ${ARGV})
+        set_property(TARGET dispatch_objects PROPERTY POSITION_INDEPENDENT_CODE ON)
+        add_library(dispatch SHARED $<TARGET_OBJECTS:dispatch_objects>)
+        set_target_properties(dispatch PROPERTIES
+            LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib/swift/${SWIFT_OS}
+            ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib/swift/${SWIFT_OS}
+        )
+        add_library(dispatch_static STATIC $<TARGET_OBJECTS:dispatch_objects>)
```

Comment:
This breaks for windows.  Please do a combined build for Windows.  We need to build the objects twice, using the same objects will not work.

Reply:
No, object libraries are fine, just as long as they are built for the final type (e.g. built with knowledge that they will be used for a shared library or not).  The best way to accomplish this is to perform two builds with the right CMake flags so that you build the objects the right way.